apps/vmq_server/src/vmq_message_store.erl

```diff
@@ -1,161 +1,103 @@
-    ]).
-delete_with_retry(SubscriberId, MsgRef, N) when N >= 0 ->
-    case vmq_plugin:only(msg_store_delete, [SubscriberId, MsgRef]) of
-        {ok, _Count} ->
-            ok;
-        {error, no_matching_hook_found} = ErrRes ->
-            ErrRes;
-        {error, Err} ->
-            lager:error("Error: ~p", [Err]),
-            vmq_metrics:incr_msg_store_ops_error(delete),
-            timer:sleep(?RETRY_INTERVAL),
-            delete_with_retry(SubscriberId, MsgRef, N - 1)
-    end;
-delete_with_retry(_SId, _MsgRef, _N) ->
-    vmq_metrics:incr_msg_store_retry_exhausted(delete),
-    ok.
+delete(SubscriberId, _MsgRef) ->
+    vmq_redis:query(
+        vmq_message_store_redis_client,
+        ["LPOP", term_to_binary(SubscriberId), 1],
+        ?LPOP,
+        ?MSG_STORE_DELETE
+    ).
 
 find(SubscriberId) ->
-    vmq_util:timed_measurement({?MODULE, find}, ?MODULE, find_with_retry, [
-        SubscriberId, ?NR_OF_RETRIES
-    ]).
-find_with_retry(SubscriberId, N) when N >= 0 ->
-    case vmq_plugin:only(msg_store_find, [SubscriberId]) of
-        {ok, _} = OkRes ->
-            OkRes;
-        {error, no_matching_hook_found} = ErrRes ->
-            ErrRes;
-        {error, Err} ->
-            lager:error("Error: ~p", [Err]),
-            vmq_metrics:incr_msg_store_ops_error(find),
-            timer:sleep(?RETRY_INTERVAL),
-            find_with_retry(SubscriberId, N - 1)
-    end;
-find_with_retry(_SId, _N) ->
-    vmq_metrics:incr_msg_store_retry_exhausted(find),
-    {error, retry_exhausted}.
+    case
+        vmq_redis:query(
+            vmq_message_store_redis_client,
+            ["LRANGE", term_to_binary(SubscriberId), "0", "-1"],
+            ?FIND,
+            ?MSG_STORE_FIND
+        )
+    of
+        {ok, MsgsInB} ->
+            DMsgs = lists:foldr(
+                fun(MsgB, Acc) ->
+                    Msg = binary_to_term(MsgB),
+                    D = #deliver{msg = Msg, qos = Msg#vmq_msg.qos},
+                    [D | Acc]
+                end,
+                [],
+                MsgsInB
+            ),
+            {ok, DMsgs};
+        Res ->
+            Res
+    end.
+
+%% ===================================================================
+%% Supervisor callbacks
+%% ===================================================================
+
+-spec init([]) ->
+    {'ok',
+        {{'one_for_one', 5, 10}, [
+            {atom(), {atom(), atom(), list()}, permanent, pos_integer(), worker, [atom()]}
+        ]}}.
+init([]) ->
+    StoreCfgs = application:get_env(vmq_server, message_store, [
+        {redis, [
+            {connect_options, "[{sentinel, [{endpoints, [{\"localhost\", 26379}]}]},{database,2}]"}
+        ]}
+    ]),
+    Redis = proplists:get_value(redis, StoreCfgs),
+
+    {ok,
+        {{one_for_one, 5, 10}, [
+            {eredis,
+                {eredis, start_link, [
+                    [
+                        {username, proplists:get_value(username, Redis)},
```

Comment:
convert `atom` to `list` according to
apps/vmq_generic_offline_msg_store/src/engines/vmq_offline_storage_engine_redis.erl